zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/DataSetInfo.scala

```diff
@@ -13,3 +17,125 @@ case class DataSetInfo(name: String,
                        schema: Schema,
                        dataInterval: Interval,
                        stats: Stats)
+
+object DataSetInfo {
+
+  def parse(json: JsValue): DataSetInfo = {
+    json.validate[DataSetInfo] match {
+      case js: JsSuccess[DataSetInfo] => js.get
+      case e: JsError => throw JsonRequestException(JsError.toJson(e).toString())
+    }
+  }
+
+  def write(dataSetInfo: DataSetInfo): JsValue = Json.toJson(dataSetInfo)
+
+  implicit val intervalFormat: Format[Interval] = new Format[Interval] {
+    override def reads(json: JsValue) = {
+      val start = (json \ "start").as[DateTime]
+      val end = (json \ "end").as[DateTime]
+      JsSuccess(new Interval(start.getMillis, end.getMillis))
+    }
+
+    override def writes(interval: Interval): JsValue = {
+      JsObject(List("start" -> Json.toJson(interval.getStart), "end" -> Json.toJson(interval.getEnd)))
+    }
+  }
+
+  implicit def tuple2Writes: Writes[Tuple2[String, String]] = Writes[(String, String)](t => Json.obj("level" -> t._1, "field" -> t._2))
+
+  def parseLevels(levelSeq: Seq[Map[String, String]]): Seq[(String, String)] = {
+    levelSeq.map {
+      levelMap => (levelMap.get("level").getOrElse(""), levelMap.get("field").getOrElse(""))
```

Comment:
better to throws than use default value `""`?  let's just use `.get`, it will throw error if the key doesn't exsit.